broadinstitute/scp-ingest-pipeline
# Store ontology label in BigQuery for synonyms in metadata (SCP-3987)

Storing synonyms in BigQuery generates "clutter" in Advanced search facet filters. Previously manual cleanup has tried but synonym storage allows "clutter" to re-accumulate. This PR aims to remove the root cause of some of the "clutter" in facet filters (The other contributing cause is change from upstream ontologies, which is less frequent).

To test:

1. In your local instance, set up this docker container to perform ingest:
`gcr.io/broad-singlecellportal-staging/scp-ingest-jlc_synonym_mitigation:d7b1747`

2. ingest metadata file from single_cell_portal_core repo:
> db/seed/synthetic_studies/array_mouse_eye/metadata.tsv

3. Confirm in study details for the study, that disease__ontology_label had arrays with synonyms like:
astigmatism (disease)
cataract (disease)
glaucoma (disease)

4. Go to BigQuery via GCP console and check the stored ontology labels by running this query:
```
SELECT DISTINCT dz_ol, 
	        FROM cell_metadata_development.alexandria_convention as c,
	        UNNEST(c.disease__ontology_label) AS dz_ol
	WHERE c.study_accession = "SCPXX"
```
Identify the study in your local instance that you used for upload and replace XX with the accession

Confirm dz_ol do not have any synonyms (ie have "(disease)" appended to the term):
cataract
astigmatism
glaucoma
HIV infectious disease
